### PR TITLE
[CRN-1298] add relatedResearch field to Research Output

### PR DESCRIPTION
**Pull request by ricardocarvalh0**

### CODE REVIEW

**Review by oratiu — CHANGES_REQUESTED** [review]
The FE looks good, only thing missing is a story for the new component in Storybook.

File: apps/crn-server/src/entities/research-output.ts [hunk 1]
```diff
@@ -126,6 +130,28 @@ const parseGraphqlTeamLite = (
...
 
+const parseGraphqlResearchOutputLite = ({
+  id: researchOutputId,
+  flatData,
+}: FetchResearchOutputRelatedResearch): Pick<
+  ResearchOutputDataObject,
+  'id' | 'title' | 'type' | 'documentType' | 'teams'
+> => ({
+  id: researchOutputId,
+  title: flatData?.title || '',
```
Comment (gabiayako) [inline apps/crn-server/src/entities/research-output.ts:141]:
```suggestion
  title: flatData.title || '',
```

File: apps/crn-server/src/entities/research-output.ts [hunk 2]
```diff
@@ -126,6 +130,28 @@ const parseGraphqlTeamLite = (
...
+  type: researchOutputMapType(flatData?.type) || undefined,
```
Comment (gabiayako) [inline apps/crn-server/src/entities/research-output.ts:142]:
```suggestion
  type: researchOutputMapType(flatData.type) || undefined,
```

File: apps/crn-server/src/entities/research-output.ts [hunk 3]
```diff
@@ -126,6 +130,28 @@ const parseGraphqlTeamLite = (
...
+  documentType:
+    flatData?.documentType &&
```
Comment (gabiayako) [inline apps/crn-server/src/entities/research-output.ts:144]:
```suggestion
    flatData.documentType &&
```

File: packages/model/src/research-output.ts [hunk 4]
```diff
@@ -218,6 +218,12 @@ export type ResearchOutputDataObject = ResearchOutputCoreObject & {
   subtype?: string;
   teams: Pick<TeamResponse, 'id' | 'displayName'>[];
   workingGroups: Pick<WorkingGroupResponse, 'id' | 'title'>[];
+  relatedResearch?: Array<
```
Comment (gabiayako) [inline packages/model/src/research-output.ts:221]:
```suggestion
  relatedResearch: Array<
```

I believe we can remove this `?` as in the parser (https://github.com/yldio/asap-hub/blob/crn-1298-related-research-to-shared-output/apps/crn-server/src/entities/research-output.ts#L67:L70) we guarantee that `relatedResearch` is always returned, right?

File: packages/react-components/src/organisms/ResearchOutputRelatedResearchCard.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  relatedResearch,
+  getRelatedResearchSuggestions = noop,
+  onChangeRelatedResearch = noop,
+  isSaving,
+  isEditMode,
+}) => (
+  <FormCard title="Are there any related outputs?">
+    <LabeledMultiSelect<ResearchOutputOption>
+      title="Related Outputs"
+      description="List all resources that were important to the creation of this output. Only published outputs on the CRN Hub will be available below."
```
Comment (gabiayako) [inline packages/react-components/src/organisms/ResearchOutputRelatedResearchCard.tsx:52]: This looks different from figma. Maybe this text "List all ..." should not be part of this `LabeledMultiSelect`, maybe it should be a text above this `LabeledMultiSelect` component and here we wouldn't have a description.
Screenshot 2023-03-08 at 14 29 50

**Review by gabiayako — CHANGES_REQUESTED** [review]
Minor comments, but I am requesting changes to take a second look later :)

File: packages/react-components/src/molecules/FormCard.tsx [hunk 6]
```diff
@@ -1,16 +1,36 @@
...
 }
 
-const FormCard: React.FC<FormCardProps> = ({ children, title }) => (
+const descriptionStyles = css({
+  paddingTop: 0,
+  paddingBottom: 0,
+  'p:first-of-type': {
+    marginTop: 0,
+    marginBottom: `${10 / perRem}em`,
+  },
```
Comment (gabiayako) [inline packages/react-components/src/molecules/FormCard.tsx:17]: is this necessary? 🤔

File: packages/react-components/src/templates/__tests__/ResearchOutputForm.test.tsx [hunk 7]
```diff
@@ -375,6 +394,23 @@ describe('on submit', () => {
...
+
+  it('can submit a related research', async () => {
+    getRelatedResearchSuggestions.mockResolvedValue([
+      { label: 'First Related Research', value: '1' },
+      { label: 'Second Related Research', value: '2' },
+    ]);
+    await setupForm();
+
+    userEvent.click(screen.getByRole('textbox', { name: /Related Outputs/i }));
+    // await waitForElementToBeRemoved(() => screen.queryByText(/loading/i));
```
Comment (gabiayako) [inline packages/react-components/src/templates/__tests__/ResearchOutputForm.test.tsx:406]: Can we remove this comment? :)

File: packages/react-components/src/templates/SharedResearchOutput.tsx [hunk 8]
```diff
@@ -119,6 +119,9 @@ const SharedResearchOutput: React.FC<SharedResearchOutputProps> = ({
             </div>
           </Card>
         )}
+        {/* TODO:  {!isGrantDocument && (
+          <RelatedResearch relatedResearch={props.relatedResearch} />
+        )} */}
```
Comment (gabiayako) [inline packages/react-components/src/templates/SharedResearchOutput.tsx:124]: was intentional to leave this commented out? I believe this is part of @oratiu task, right?

**Review by gabiayako — APPROVED** [review]

**Review by oratiu — APPROVED** [review]

**Pull request closed without merging**